core/src/main/java/nl/inl/blacklab/index/config/DocIndexerXPath.java

```diff
@@ -29,15 +29,20 @@
 import com.ximpleware.XPathParseException;
 
 import nl.inl.blacklab.index.Indexer;
+import static nl.inl.blacklab.index.config.DocIndexerConfig.replaceDollarRefs;
 import nl.inl.blacklab.index.config.InlineObject.InlineObjectType;
 import nl.inl.util.ExUtil;
 import nl.inl.util.StringUtil;
 import nl.inl.util.XmlUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /**
  * An indexer configured using full XPath 1.0 expressions.
  */
 public class DocIndexerXPath extends DocIndexerConfig {
+    
+    private static final Logger LOGGER = LogManager.getLogger(DocIndexerXPath.class);
```

Comment:
Let's stick with lowercase 'logger' like it's used everywhere else. In my (admittedly arbitrary) opinion, only constant values should be uppercased. While this variable is final, it doesn't represent a constant value but a logging system.

Reply:
fine

Reply:
though I think the rest of the world and static code analysis tools say all static final fields should be uppercase

Reply:
I don't think there's a broad consensus on that. Here's [an alternative viewpoint](https://stackoverflow.com/a/18640239), for example. And static code analysis tools tend to stick to rules that are easily checked, which are not necessarily the ones that are the clearest for a human reading the code.

Anyway, this is not really the stuff that matters. Although many holy wars are fought over them, of course. :-)